drv/transceivers-server/src/main.rs

```diff
@@ -160,16 +187,87 @@ impl idl::InOrderTransceiversImpl for ServerImpl {
     }
 }
 
+impl NotificationHandler for ServerImpl {
+    fn current_notification_mask(&self) -> u32 {
+        TIMER_NOTIFICATION_MASK
+    }
+
+    fn handle_notification(&mut self, _bits: u32) {
+        // Check for errors
+        let errors = self.leds.error_summary().unwrap();
+        if errors != self.led_error {
+            self.led_error = errors;
+            ringbuf_entry!(Trace::LEDErrorUpdate(errors))
+        }
+
+        // Query module presence and update LEDs accordingly
+        let presence = match self.transceivers.get_modules_status() {
+            Ok(status) => status.present,
+            Err(_) => 0,
+        };
+
+        if presence != self.modules_present {
+            // Errors are being temporarily suppressed here due to a miswiring of
+            // the I2C bus at the LED controller parts. They will not be accessible
+            // without rework to older hardware, and newer (correct) hardware will
+            // be replacing the hold stuff very soon.
+            // TODO: remove error suppression here once Rev B hardware is in
+            let _ = self.leds.update_led_state(presence);
+            self.modules_present = presence;
+            ringbuf_entry!(Trace::ModulePresenceUpdate(presence));
+        }
+
+        let next_deadline = sys_get_timer().now + TIMER_INTERVAL;
+        sys_set_timer(Some(next_deadline), TIMER_NOTIFICATION_MASK)
+    }
+}
+
 #[export_name = "main"]
 fn main() -> ! {
     loop {
-        let mut buffer = [0; idl::INCOMING_SIZE];
+        // This is a temporary workaround that makes sure the FPGAs are up
+        // before we start doing things with them. A more sophisticated
+        // notification system will be put in place.
```

Comment:
yup, don't like this 😆

Reply:
More helpfully, could you use the `Sequencer.front_io_phy_ready` API (like [this](https://github.com/oxidecomputer/hubris/blob/master/task/monorail-server/src/bsp/sidecar_b.rs#L138-L146)) instead of manually recreating it here?

Reply:
Yeah, me either! We should chat about this, with @arjenroodselaar if possible.

Reply:
Dang, I had this window open so I never saw your second comment before I responded. I'll push another commit which addresses this.